File: taekwondo Spring/taekwondo/src/main/java/com/taekwondo/service/UsuarioService.java
package com.taekwondo.service;

import java.util.List;

import org.springframework.http.ResponseEntity;

import com.taekwondo.model.Usuario;

public interface UsuarioService {
	public abstract List<Usuario> getUsuarios();
	public abstract Usuario getUsuario(String username);
	public abstract ResponseEntity<Object> createUsuario(Usuario usuario);
	public abstract void updateUsuario(String username, Usuario usuario);
	public abstract void deleteUsuario(String username);
}

File: proyectoTaeKwonDo/src/app/_services/usuario.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Usuario } from '../_models/usuario';

@Injectable({
  providedIn: 'root'
})
export class UsuarioService {
  API_URI = 'http://localhost:8080';

  constructor(private http: HttpClient) { }

  getUsuarios(){
    return this.http.get(this.API_URI+'/usuario');
  }

  getUsuario(usuario: string){
    return this.http.get(this.API_URI+'/usuario/'+usuario);
  }

  createUsuario(usuario: Usuario){
    return this.http.post(this.API_URI+'/usuario', usuario);
  }

  updateUsuario(usuario: Usuario){
    return this.http.put(this.API_URI+'/usuario/'+ usuario.usuario, usuario);
  }

  deleteUsuario(usuario: string){
    return this.http.delete(this.API_URI+'/usuario/'+usuario);
  }
}
